src/Gherkin.php

```diff
@@ -106,12 +106,14 @@ public function load($resource, array $filters = [])
         $filters = array_merge($this->filters, $filters);
 
         $matches = [];
-        if (preg_match('/^(.*):(\d+)-(\d+|\*)$/', $resource, $matches)) {
-            $resource = $matches[1];
-            $filters[] = new LineRangeFilter($matches[2], $matches[3]);
-        } elseif (preg_match('/^(.*):(\d+)$/', $resource, $matches)) {
-            $resource = $matches[1];
-            $filters[] = new LineFilter($matches[2]);
+        if (is_scalar($resource) || $resource instanceof \Stringable) {
```

Comment:
(PHPStan) `$resource` is not always stringable.

I'm checking for `scalar` or `Stringable` for widest compatibility possible - is still a BC concern?